plugins/scripts/upload/trns_transform_mzML_LCMS_to_MetaboliteAtlas2_MAFileInfo.py

```diff
@@ -0,0 +1,228 @@
+        working_directory: The directory the resulting json file will be
+                           written to.
+        shock_id: Shock id for the hdf file if it already exists in shock
+        handle_id: Handle id for the hdf file if it already exists as a
+                    handle
+        input_mapping: JSON string mapping of input files to expected types.
+                       If you don't get this you need to scan the input
+                       directory and look for your files.
+        level: Logging level, defaults to logging.INFO.
+        atlases: List of MetaboliteAtlas atlas IDs.
+        name: Name of the file, optional.  Defaults to the file name.
+        polarity: Run polarity.
+        group: Run group.
+        inclusion_order: Run inclusion_order.
+        retention_correction: Run retention_correction.
+        normalization_factor: Run normalization factor.
+
+    Returns:
+        JSON files on disk that can be saved as a KBase workspace objects.
+
+    Authors:
+        Steven Silvester
+    """
+
+    if logger is None:
+        logger = script_utils.stderrlogger(__file__)
+
+    logger.info("Starting conversion of mzML to MetaboliteAtlas2.MAFileInfo")
+    token = os.environ.get('KB_AUTH_TOKEN')
+
+    if not working_directory or not os.path.isdir(working_directory):
+        raise Exception("The working directory {0} is not a valid directory!"
+                        .format(working_directory))
+
+    logger.info("Scanning for mzML files.")
+
+    valid_extensions = [".mzML"]
+
+    files = os.listdir(input_directory)
+    mzml_files = [x for x in files
+                             if os.path.splitext(x)[-1] in valid_extensions]
+
+    assert len(mzml_files) != 0
+
+    logger.info("Found {0} files".format(len(mzml_files)))
+
+    for fname in mzml_files:
+        path = os.path.join(input_directory, fname)
+
+        if not os.path.isfile(path):
+            raise Exception("The input file name {0} is not a file!"
+                            .format(path))
+
+        hdf_file = mzml_loader.mzml_to_hdf(path)
+
+        if shock_service_url:
+            shock_info = script_utils.upload_file_to_shock(logger,
+                    shock_service_url, hdf_file, token=token)
+
+        run_info = dict()
+        run_info['name'] = name or fname.replace('.mzML', '')
+        run_info['atlases'] = atlases or []
+        if polarity is not None:
+            run_info['polarity'] = polarity
+        if group is not None:
+            run_info['group'] = group
+        if inclusion_order is not None:
+            run_info['inclusion_order'] = inclusion_order
+        if normalization_factor is not None:
+            run_info['normalization_factor'] = normalization_factor
+        if retention_correction is not None:
+            run_info['retention_correction'] = retention_correction
+
+        if shock_service_url:
+            run_info["run_file_id"] = shock_info["id"]
```

Comment:
And one more fix. Instead of putting shock node id into handle field you should create handle pointing to that shock node and use id of this handle in workspace object. Can you substitute this line:
```
            run_info["run_file_id"] = shock_info["id"]
```
by following lines:
```
            handle_id = script_utils.getHandles(logger, shock_service_url,
                    handle_service_url, [shock_info["id"]], token=token)[0]
            run_info["run_file_id"] = handle_id
```
?